### ISSUE TITLE
Publication fails for Camtrap DP

### ISSUE THREAD

**Issue by peterdesmet** [gbif/ipt#1953]
Publishing a Camtrap DP (ipt3.gbif-uat.org) currently fails:

> Publishing version #1.0 of resource camtrap-dp-0-6 failed: Archive generation for resource camtrap-dp-0-6 failed: io.frictionlessdata.tableschema.exception.TableValidationException: Declared column observationID not found in data

The `event-observations` file does contain a `observationID` column however. Maybe there is some code checking for an `observations` resource (Camtrap DP 0.4), while Camtrap 0.6 expects a `event-observations` and/or `media-observations` resource?

```
20:00:50 Data package generation started for version #1.0
20:00:50 Start writing data file for event-observations
20:00:51 Data file written for event-observations with 39 records and 18 columns
20:00:51 Start writing data file for media-observations
20:00:51 Data file written for media-observations with 0 records and 19 columns
20:00:51 Start writing data file for deployments
20:00:51 Data file written for deployments with 4 records and 23 columns
20:00:51 Start writing data file for media
20:00:51 Data file written for media with 423 records and 12 columns
20:00:51 All data files completed
20:00:51 Metadata added
```

**Comment by mike-podolskiy90** [comment]
Thank you for reporting the bug @peterdesmet 
Maybe the IPT expects all four entities mapped? Have you tried that?

**Comment by mike-podolskiy90** [comment]
Yes, my guess was correct. If you map all four of them you can publish the resource

**Comment by peterdesmet** [comment]
Aha! The intended behaviour is that all of the following are valid:

- [ ]  `deployments`, `media`, `media-observations`
- [ ]  `deployments`, `media`, `event-observations`
- [ ]  `deployments`, `media`, `media-observations`, `event-observations`

**Comment by mike-podolskiy90** [comment]
Ok, if we want more complex functionality there we need to implement it somehow in the schema file https://rs.gbif.org/sandbox/experimental/schemas.json , so IPT can understand what combinations are valid

**Comment by peterdesmet** [comment]
Thanks! Suggestions on how to express this best in `schemas.json`? Ping @timrobertson100.

**Comment by mike-podolskiy90** [comment]
Add something like this to https://rs.gbif.org/sandbox/experimental/schemas.json at the scheme level

```
{
  "constraints": [
    {
      "required": ["deployments", "media"],
      "anyOf": [
        {"required": ["media-observations"]},
        {"required": ["event-observations"]}
      ]
    }
  ]
}
```

**Comment by mike-podolskiy90** [comment]
keys would be `required`, `optional`, `anyOf`, `oneOf`

**Comment by mike-podolskiy90** [comment]
I've updated rs.gbif.org by adding schema requirements. Just deployed and tested on ipt3.gbif-uat.org, looks fine
